### PR TITLE
Add `CircuitStatus` to `CircuitPredicate`

### PR DESCRIPTION
**Pull request by shannynalayna**
Adds a `CircuitStatus` circuit predicate to allow for filtering circuits and proposals based on the `circuit_status` field. This also affects the circuits returned by the admin store's `list_circuits` operation, as only `Active` circuits are returned by default. 

The `CircuitStatus` predicate is also added to the `/admin/proposals` and `/admin/circuits/` endpoints, via the `status` query parameter. 

Also adds a `circuit-status` option to the `splinter circuit list`, `splinter circuit proposals` commands. 

Also makes a small update to the data displayed via the `show` command, to display the `circuit_status` field of the circuit / proposed circuit.

### CODE REVIEW

File: cli/src/action/circuit/api.rs [hunk 1]
```diff
@@ -245,18 +246,25 @@ pub struct CircuitSlice {
...
+    pub circuit_status: Option<CircuitStatus>,
 }
 
 impl fmt::Display for CircuitSlice {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let mut display_string = format!("Circuit: {}\n    ", self.id,);
 
         if let Some(display_name) = &self.display_name {
-            display_string += &format!("Display Name: {}\n    ", display_name);
+            display_string += &format!("Display Name: {:?}\n    ", display_name);
```
Comment (agunde406) [inline cli/src/action/circuit/api.rs:257]: Was this change intentional?

Comment (shannynalayna) [reply cli/src/action/circuit/api.rs]: Oh, no it was not. Fixed

File: cli/src/action/circuit/api.rs [hunk 2]
```diff
@@ -257,6 +259,12 @@ impl fmt::Display for CircuitSlice {
             display_string += "Display Name: -\n    ";
         }
 
+        if let Some(status) = &self.circuit_status {
+            display_string += &format!("Circuit Status: {:?}\n    ", status);
```
Comment (peterschwarz) [inline cli/src/action/circuit/api.rs:263]: This should  match on status and output a string, otherwise, it will display the qualified variant, like `CircuitStatus::Active`

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: cli/src/action/circuit/api.rs (hunk 2 above)
Comment (shannynalayna) [reply cli/src/action/circuit/api.rs]: Updated to include using a `From` implementation on the `CircuitStatus` that is displayed here (from the `admin::messages` module) to convert the full variant to just `Active/Disbanded/Abandoned`, depending.

File: cli/man/splinter-circuit-show.1.md [hunk 3]
```diff
@@ -104,6 +105,42 @@ Proposal to create: 01234-ABCDE
                 AA01
 ```
 
+The information displayed below will appear the same on all proposed member
+nodes for a circuit proposal to disband an existing circuit.
+If all member nodes vote to accept disbanding the circuit, the
+`splinter-circuit-show` command will display the same information, without the
+`Vote` as all nodes would have accepted the proposal. If any of the member
+nodes vote to reject disbanding the circuit, the proposal will not be viewable
```
Comment (agunde406) [inline cli/man/splinter-circuit-show.1.md:113]: "the proposal will not be viewable" this is kinda misleading because really the proposal will no longer exist.

**Review by agunde406 — CHANGES_REQUESTED** [review]
Tested after rebasing on https://github.com/Cargill/splinter/pull/1158

File: cli/man/splinter-circuit-show.1.md (hunk 3 above)
Comment (shannynalayna) [reply cli/man/splinter-circuit-show.1.md]: Updated here to say the `the proposal will be removed and is not viewable by any nodes`, and also clarified the wording above where the same sentence was used (and the same circumstances pertain, the proposal is not only no longer viewable, but has been removed)

File: cli/man/splinter-circuit-show.1.md [hunk 4]
```diff
@@ -76,13 +76,14 @@ The information displayed below will appear the same on all proposed member node
 If all member nodes vote to accept the circuit, the `splinter-circuit-show`
 command will display the same information, without the `Vote` as all nodes would
 have accepted the proposal. If any of the member nodes vote to reject the circuit,
-the proposal will not be viewable by any nodes.
+the proposal will be removed and is not be viewable by any nodes.
```
Comment (agunde406) [inline cli/man/splinter-circuit-show.1.md:79]: " not be viewable by any nodes" -> " not viewable by any nodes"

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: cli/man/splinter-circuit-show.1.md (hunk 4 above)
Comment (shannynalayna) [reply cli/man/splinter-circuit-show.1.md]: Updated

**Review by agunde406 — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]
